### ISSUE TITLE
Access Contents Information is not respected for Navigation and Listings

### ISSUE THREAD

**Issue by agitator** [plone/Products.CMFPlone#260]
Access Contents Information:
This permission allows access to an object, without necessarily
viewing the object. For example, a user may want to see the object's
title in a list of results, even though the user can't view the
contents of that file.

Objects with this permission are neither listed in collections or listings.
Furthermore if i want a private object to appear in the navigation (for example a link to an intranet, which the requires a login) it isn't shown either.

I think the indexer should also check for this ^ permission.
https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/CatalogTool.py#L124

**Comment by vangheem** [comment]
Any update on this? Closing for now as this enhancement request is quite stale.

**Comment by agitator** [comment]
Adding
```
    allowed.update(rolesForPermissionOn('Access contents information', obj))
```
in https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/CatalogTool.py#L145 would fix this.

If something shouldn't be listed, 'Access contents information' should be turned off anyway.

-------------

I ran into this again and if no one speaks out against it, I'll fix this for 5.2 at the https://alpinecity.tirol sprint.

**Comment by agitator** [comment]
After further discussion with @jensens, we came to the conclusion that 'Access contents information' is the correct permission to check if an item should appear in listings and navigation.

Since we use 'View' and 'Access contents information' in all Plone workflows, this change shouldn't have any effect on any existing Plone site but furthermore frees up the 'View' permission to make that differentiation in more complex usecases.
